pyscf/df/incore.py

```diff
@@ -28,7 +28,9 @@
 
 
 MAX_MEMORY = getattr(__config__, 'df_outcore_max_memory', 2000)  # 2GB
-LINEAR_DEP_THR = getattr(__config__, 'df_df_DF_lindep', 1e-9)
+# LINEAR_DEP_THR cannot be below 1e-7,
+# see qchem default setting in https://manual.q-chem.com/5.4/sec_Basis_Customization.html
+LINEAR_DEP_THR = getattr(__config__, 'df_df_DF_lindep', 1e-7)
 
```

Comment:
The comment is misleading. This is the threshold for density fitting, while the Q-Chem manual refers to the linear dependence threshold for the orbital basis set.

Reply:
Note that Psi4 uses the value `1e-10` by default for the auxiliary basis.

I believe Q-Chem does not employ a linear dependency threshold, instead opting to solve the RI equations directly.

Reply:
> Note that Psi4 uses the value `1e-10` by default for the auxiliary basis.
> 
> I believe Q-Chem does not employ a linear dependency threshold, instead opting to solve the RI equations directly.

`1e-10` is probably still OK for SCF. But the numerical issues are observed in gradient and Hessian calculations where multiple inversions/solves are involved. To be consistent, I propose we use a larger threshold for the calculations. I was not sure how it affects the accuracy.

For Q-Chem, it was my best guess. I could be wrong. But solving RI equations also requires some thresholds, right? Otherwise, the equations can be ill-conditioned. I didn't find the setting anywhere else.

Reply:
There's no threshold when you do a full solve for the linear equation ${\bf S} {\bf C} = \boldsymbol{\gamma}$ for the expansion coefficients ${\bf C}$ with the auxiliary basis Coulomb overlap ${\bf S}$ from the source terms $\gamma_A = \sum_{rs} (A|rs) P_{rs}$.

Reply:
The linear dependency issue is mostly found in the RSH functional, where pyscf computes the density fitting for the LR part. This is highly singular and numerical issues are noticed with 1e-10 threshold. I think the threshold 1e-10 or 1e-7 do not impact the density fitting for SR part, the full-range Coulomb, since there are probably very few eigenvalues between 1e-10 and 1e-7. In the case of density fitting of LR, there are many eigenvectors near 1e-10. The linearly independent sets can be quite different for geometries with small differences.